### PR TITLE
Feat/lidar car

### PR DESCRIPTION
**Pull request by idale0001**
Updated opponent detection files so that the opponents are more consistently assigned the same id and color by using a greedy algorithm to find the clusters that are closest to the previous clusters.

### CODE REVIEW

**Review by NicoBartholomai — CHANGES_REQUESTED** [review]
There is a faster way to find pairs between two sets but I think since they're so small this is fine. You could consider creating your own datatype that just holds a pointer to the clusters and their respective centroids instead of trying to keep track of the same index within lists.

File: rr_evgp/src/opponent_detection/opponent_detection.cpp [hunk 1]
```diff
@@ -47,8 +52,78 @@ visualization_msgs::Marker makeMarkers(const std::vector<geometry_msgs::Point>&
...
+    int largest_id = 0;
+    std::unordered_set<int> ids;
+    for (int i = 0; i < prevCentroids.size(); i++) {
+        std::vector<double> row;
+        if (ids.find(prevMarkers.markers[i].id) == ids.end()) {
+            ids.insert(prevMarkers.markers[i].id);
+        }
+        if (prevMarkers.markers[i].id > largest_id) {
+            largest_id = prevMarkers.markers[i].id;
+        }
```
Comment (Daniel-Martin576) [inline rr_evgp/src/opponent_detection/opponent_detection.cpp:75]: max

File: rr_evgp/src/opponent_detection/opponent_detection.cpp [hunk 2]
```diff
@@ -47,8 +52,78 @@ visualization_msgs::Marker makeMarkers(const std::vector<geometry_msgs::Point>&
...
+        for (geometry_msgs::Pose centroid : centroids) {
+            double distance = euclidDistance(prevCentroids[i].position, centroid.position);
+            row.push_back(distance);
+        }
```
Comment (Daniel-Martin576) [inline rr_evgp/src/opponent_detection/opponent_detection.cpp:79]: Could do transform if wanted to be fancy, but not nessessary

Comment (Daniel-Martin576) [reply rr_evgp/src/opponent_detection/opponent_detection.cpp:79]: Const reference

**Comment by NicoBartholomai** [comment]
This will be re-implemented in ROS 2

**Pull request closed without merging**